controllers/clustergroupupgrade_controller.go

```diff
@@ -178,6 +178,53 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 		}
 	}
 
+	progressingCondition := meta.FindStatusCondition(clusterGroupUpgrade.Status.Conditions, string(utils.ConditionTypes.Progressing))
+
+	// Before starting the upgrade check that all the managed policies exist
+	var allManagedPoliciesExist bool
+	var managedPoliciesInfo policiesInfo
+
+	// doManagedPoliciesExist should only be called before the CGU has begun processing
+	if progressingCondition != nil {
```

Comment:
`managedPoliciesInfo` needs to be populated for `buildRemediationPlan` and `processManagedPolicyForUpgradeContent` to work. So `doManagedPoliciesExist` can't be executed once only. It is a mandatory part of the process. In a way you could say the validated condition doesn't mean much as we always have to redo it until the cgu is started.

Reply:
@sabbir-47 perhaps we could change the proposal to move Validated to be after precaching and backup? Otherwise it seems we would need to rearchitect quite a bit of our reconcile loop